src/main/java/graphql/execution/Execution.java

```diff
@@ -100,11 +100,11 @@ private ExecutionResult executeOperation(ExecutionContext executionContext, Obje
         ExecutionResult result;
         try {
             if (operation == OperationDefinition.Operation.MUTATION) {
-                result = mutationStrategy.execute(executionContext, parameters);
+                result = mutationStrategy.execute(executionContext, parameters).toCompletableFuture().join();
             } else if (operation == SUBSCRIPTION) {
-                result = subscriptionStrategy.execute(executionContext, parameters);
+                result = subscriptionStrategy.execute(executionContext, parameters).toCompletableFuture().join();
             } else {
-                result = queryStrategy.execute(executionContext, parameters);
+                result = queryStrategy.execute(executionContext, parameters).toCompletableFuture().join();
             }
```

Comment:
One decision we need to make is whether to use CompletionStage or CompletableFuture (or a mix of both)

Standard Java best practice says use the interface CompletionStage.  Except you can never get the value of a CompletionStage and so you have all these .toCompleteableFuture() calls

Since this code base is all about materialising values (eventually) perhaps we should use CompletableFuture?